code/game/machinery/GBPvendor.dm

```diff
@@ -0,0 +1,125 @@
+		new /datum/GBP_equipment("Reactive Decoy Armor x5",		/obj/item/clothing/suit/armor/reactive/stealth,				6000,	5),
+		new /datum/GBP_equipment("Cloaking Belt x5",		/obj/item/device/shadowcloak,									8000,	5),
+		new /datum/GBP_equipment("Chrono Suit x5",		/obj/item/clothing/suit/space/chronos,								10000,	5),
+		new /datum/GBP_equipment("WHAT HAVE YOU DONE... x5",		/obj/vehicle/space/speedbike/speedwagon,				30000,	5),
+		)
+
+/datum/GBP_equipment
+	var/equipment_name = "generic"
+	var/equipment_path = null
+	var/cost = 0
+	var/amount = 0
+
+/datum/GBP_equipment/New(name, path, cost, amount)
+	src.equipment_name = name
+	src.equipment_path = path
+	src.cost = cost
+	src.amount = amount
+
+/obj/machinery/GBP_vendor/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/GBP_vendor/update_icon()
+	if(powered())
+		icon_state = initial(icon_state)
+	else
+		icon_state = "[initial(icon_state)]-off"
+
+/obj/machinery/GBP_vendor/attack_hand(mob/user)
+	if(..())
+		return
+	interact(user)
+
+/obj/machinery/GBP_vendor/interact(mob/user)
+	GBPearned = GBP + GBPspent
+	var/dat
+	dat +="<div class='statusDisplay'>"
+	dat += "You have <td>[GBP]</td> engineering voucher points<br>"
+	switch(round(GBPearned*100/world.time))
+		if(0 to 5)
+			dat += "Rating: Terrible<br>"
+		if(6 to 10)
+			dat += "Rating: Bad<br>"
+		if(11 to 15)
+			dat += "Rating: Subpar<br>"
+		if(16 to 20)
+			dat += "Rating: Decent<br>"
+		if(21 to 25)
+			dat += "Rating: Robust<br>"
+		if(26 to 50)
+			dat += "Rating: Good boy<br>"
+		if(51 to 999)
+			dat += "Rating: Holy Shit<br>"
+	dat += "</div>"
+	dat += "<br><b>Equipment point cost list:</b><BR><table border='0' width='300'>"
+	for(var/datum/GBP_equipment/prize in prize_list)
+		dat += "<tr><td>[prize.equipment_name]</td><td>[prize.cost]</td><td><A href='?src=\ref[src];purchase=\ref[prize]'>Purchase</A></td></tr>"
+	dat += "</table>"
+
+	var/datum/browser/popup = new(user, "vending", "Engineering Point Redemption", 400, 350)
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/GBP_vendor/Topic(href, href_list)
+	if(..())
+		return
+	if(href_list["purchase"])
+		var/datum/GBP_equipment/prize = locate(href_list["purchase"])
+		if (!prize || !(prize in prize_list))
+			return
+		if(prize.cost > GBP)
+		else if(prize.cost >= 300) // Placeholder spaghetti calm your shit
+			GBP -= prize.cost
+			GBPspent += prize.cost
+			for(var/i in 1 to prize.amount)
+				new prize.equipment_path(get_turf(src))
+			if(prize.cost== 10000) // Still a placeholder
+				for(var/i in 1 to prize.amount)
+					new /obj/item/clothing/head/helmet/space/chronos(get_turf(src))
+			feedback_add_details("Engi_equipment_bought",
+				"[src.type]|[prize.equipment_path]")
+		else
+			GBP -= prize.cost
+			new prize.equipment_path(src.loc)
+			feedback_add_details("Engi_equipment_bought",
+				"[src.type]|[prize.equipment_path]")
+	updateUsrDialog()
+
+/obj/machinery/GBP_vendor/attackby(obj/item/I, mob/user, params)
+	return ..()

```

Comment:
I kind of left it there so I wouldn't forget about giving it deconstruction options.